### DISCUSSION TITLE
Query customer by ID  SFCC

### DISCUSSION THREAD

**Discussion by rpetraglia** [SalesforceCommerceCloud/sfcc-ci#441] [Team Posts]
Hi,
it is possibile search a customer by ID SFCC?
Thank's in advance.
Regards,

**Comment by zlatinz** [comment]
https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/DWAPI/scriptapi/html/api/class_dw_customer_CustomerMgr.html

static getCustomerByCustomerNumber(customerNumber : String) : Customer
Returns the customer with the specified customer number.
static getCustomerByLogin(login : String) : Customer
Returns the customer for the specified login name.
static queryProfile(queryString : String, args : Object...) : Profile
Searches for a single profile instance.

static queryProfiles(queryString : String, sortString : String, args : Object...) : SeekableIterator
Searches for profile instances.
static queryProfiles(queryAttributes : Map, sortString : String) : SeekableIterator
Searches for profile instances.

**Comment by rpetraglia** [comment]
Hi,
I already saw this but I have only CustomerID and not customer number and there are not api to get profile by CustomerID.
Thanks

**Comment by rohanmallick08** [comment]
You can use OCAPI customer shop resource.

/customers/{customer_id}?expand={String}

https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/OCAPI/19.5/shop/Resources/Customers.html?cp=0_12_3_5
